### ISSUE TITLE
Capture showing wrong cursor with multiple Captures trait with differnt cursors against different targets.

### ISSUE THREAD

**Issue by MustaphaTR** [OpenRA/OpenRA#21055]
https://github.com/OpenRA/OpenRA/pull/20261 seem to have regressed something. I've engine updated GenAlpha to include it and now building capturing use the enter cursor instead of ability one. The enter cursor seems to be coming from the Capture trait for driverless vehicles, commenting out that Captures: trait seems to show the ability cursor properly for the capturer infantry.

**Comment by PunkPun** [comment]
Cursor priority shouldn't have changed?

**Comment by MustaphaTR** [comment]
I couldn't figure out why it happens, but it does. Seem to visible in basemods (TD) too, with my testcase screenshot on that PR. Trying to repair an enemy husk still shows a red cursor provided its health is above the threshold, while it should be always green. You can also test by adding `EnterCursor: ability` to the first Captures, and see that all 3 cases start using it.

**Comment by MustaphaTR** [comment]
This seem to be affecting Voicelines too.

I think i can see why it is happening now, tho not sure how to fix it. [Here](https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Mods.Common/Traits/Captures.cs#L132) the CanTarget check returns true for all Captures traits on the actor, making the first trait used for the order generator, as such for cursor and voicelines, regardless of that specific trait's validness of the current target.

**Comment by PunkPun** [comment]
Yes, `CaptureManager.CanTarget` is supposed to be used in aggregate by classes that don't care which `Captures` is selected. While in this case it's for individual `Captures`

I fixed the original issue with selection like this [here like this](https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Mods.Common/Traits/CaptureManager.cs#L163)

**Comment by PunkPun** [comment]
> Trying to repair an enemy husk still shows a red cursor provided its health is above the threshold

It shows green cursor on bleed

**Comment by MustaphaTR** [comment]
> It shows green cursor on bleed

Cannot confirm, are you sure the husk you tested with didn't already have less than 50% health? After that point, which natually happens after some time, it does turn to green. It should always be green instead, regardless of the health.

**Comment by penev92** [comment]
Is this even actionable for the current release cycle? Because it is on the milestone.
